### PR TITLE
RBAC: Only display unique permissions in list

### PR DESCRIPTION
**Pull request by kalleep**
**What this PR does / why we need it**:
When permissions get displayed in the acl list there can be duplications. E.g. Admin has edit on dashboards but inherits Admin from folder. 

With this pr we only render the highest permission a user / team / role has.

Before change:

After change:

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/3289

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/AccessControl/PermissionList.tsx [hunk 1]
```diff
@@ -1,19 +1,36 @@
...
+  const computed = useMemo(() => {
+    const keep: { [key: string]: ResourcePermission } = {};
+    for (let item of items) {
+      const key = item[compareKey]!;
+      if (!keep[key]) {
+        keep[key] = item;
+        continue;
+      }
+
+      if (item.actions.length > keep[key].actions.length) {
```
Comment (IevaVasiljeva) [inline public/app/core/components/AccessControl/PermissionList.tsx:26]: comment: this feels a little bit cheeky. I think there could be a case in the future where user friendly permissions are not necessarily subsets of one another. But it works for now, so let's just go with it :)

File: public/app/core/components/AccessControl/Permissions.tsx [hunk 2]
```diff
@@ -102,23 +102,23 @@ export const Permissions = ({
     () =>
       sortBy(
         items.filter((i) => i.teamId),
-        ['team']
+        ['team', 'isManaged']
```
Comment (IevaVasiljeva) [inline public/app/core/components/AccessControl/Permissions.tsx:105]: question: what difference does this change make?

**Review by IevaVasiljeva — APPROVED** [review]
Nice one! I like how simple this solution is :)

File: public/app/core/components/AccessControl/PermissionList.tsx (hunk 1 above)
Comment (kalleep) [reply public/app/core/components/AccessControl/PermissionList.tsx:26]: Yes I know and we should look into how rework this ui to allow more control (i.e. not use friendly permissions)

File: public/app/core/components/AccessControl/Permissions.tsx (hunk 2 above)
Comment (kalleep) [reply public/app/core/components/AccessControl/Permissions.tsx:105]: Non managed permissions will be sorted before managed and if we have equal action length the non managed will be displayed

**Pull request merged**